### PR TITLE
NDRS 365: Remove Lint Exceptions

### PR DESCRIPTION
**Pull request by darthsiroftardis**
Removed all lint exceptions for `allow(unused)` and removed all but two exceptions for `allow(dead_code)`. Two enums have one used variant each, these unused variants are being used in tests. Therefore the `allow(dead_code)` persists in those two cases.

### CODE REVIEW

File: node/src/components/consensus/tests/consensus_des_testing.rs [hunk 1]
```diff
@@ -148,7 +148,7 @@ where
 
 pub(crate) enum DeliverySchedule {
     AtInstant(Timestamp),
-    #[allow(dead_code)] // TODO: Use or remove this.
+    #[allow(dead_code)] // Drop variant used in tests.
```
Comment (afck) [inline node/src/components/consensus/tests/consensus_des_testing.rs:151]: Isn't the whole module `#[cfg(test)]`? It shouldn't be complaining about dead code if it's actually used in tests then.

File: node/src/components/linear_chain_sync.rs [hunk 2]
```diff
@@ -162,9 +162,8 @@ pub(crate) struct LinearChainSync<I> {
 }
 
 impl<I: Clone + PartialEq + 'static> LinearChainSync<I> {
-    #[allow(unused)]
     pub fn new<REv: ReactorEventT<I>>(
-        effect_builder: EffectBuilder<REv>,
+        _effect_builder: EffectBuilder<REv>,
```
Comment (afck) [inline node/src/components/linear_chain_sync.rs:166]: Why do we pass that parameter at all? Are we intending to use it soon? (@goral09?)

File: node/src/components/storage/lmdb_store.rs [hunk 3]
```diff
@@ -14,8 +14,6 @@ use crate::types::json_compatibility::ExecutionResult;
 #[repr(u8)]
 enum Tag {
     // TODO - remove once used.
-    #[allow(unused)]
```
Comment (afck) [inline node/src/components/storage/lmdb_store.rs:17]: The comment above should be removed, too.

**Review by afck — APPROVED** [review]
Looks good to me, although I can't really comment on most of the code. Would be good if someone more familiar with the execution engine, API server, storage and smart contracts could also take a look.

On the one hand I hope we're not getting in the way of anyone's work in progress. On the other hand I think in general dead code just shouldn't be committed.

File: execution_engine/src/storage/trie_store/operations/mod.rs [hunk 4]
```diff
@@ -553,6 +553,7 @@ enum KeysIteratorState<K, V, S: TrieStore<K, V>> {
     /// Iterate normally
     Ok,
     /// Return the error and stop iterating
+    #[allow(dead_code)] // Return variant alone is used in testing.
```
[the remark this answers is not in the document]
Comment (darthsiroftardis) [reply execution_engine/src/storage/trie_store/operations/mod.rs:556]: I think the compiler can't catch it for enums specifically, and this was the only way I could proceed

File: node/src/components/consensus/tests/consensus_des_testing.rs (hunk 1 above)
Comment (darthsiroftardis) [reply node/src/components/consensus/tests/consensus_des_testing.rs:151]: It does indeed complain, I tried to resolve it

**Pull request merged**